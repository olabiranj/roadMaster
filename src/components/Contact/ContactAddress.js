import React from "react";

const ContactAddress = () => {
  return (
    <div className="col-lg-6">
      <div className="contact-address">
        <h4>Address</h4>
        <p>
          EPIC tower, 232 Ikorodu RD, Ilupeju, Lagos. <br />
          <br />
          <span className="text-info">You can also reach us on:</span>
        </p>
        <ul>
          <li>
            <div className="contact-address-icon">
              <i className="icofont icofont-headphone-alt"></i>
            </div>
            <div className="contact-address-info">
              <a href="callto:#">+234 8012 435 941</a>
              <a href="callto:#">+234 9041 800 937</a>
            </div>
          </li>
          <li>
            <div className="contact-address-icon">
              <i className="icofont icofont-envelope"></i>
            </div>
            <div className="contact-address-info">
              <a href="mailto:#">team-003gmail.com</a>
            </div>
          </li>
          <li>
            <div className="contact-address-icon">
              <i className="fa fa-github"></i>
            </div>
            <div className="contact-address-info">
              <a href="#s">www.github.com/buildforSDG-003.com</a>
            </div>
          </li>
        </ul>
      </div>
    </div>
  );
};

export default ContactAddress;
