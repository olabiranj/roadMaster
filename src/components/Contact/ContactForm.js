import React, { useState } from "react";

const ContactForm = () => {
  let [name, setName] = useState("");
  let [email, setEmail] = useState("");
  let [subject, setSubject] = useState("");
  let [message, setMessage] = useState("");
  let [error, setError] = useState("");
  let [msg, setMsg] = useState("");

  const submitQuery = (event) => {
    event.preventDefault();
    if (message.length < 15) {
      setError('*Message Too Short!')
    }
    if (subject.length < 5) {
      setError('*Subject Too Short!')
    }
    setMsg('Message Sent')
  };
  return (
    <div className="col-lg-6">
      <div className="contact-form">
        <h4>Get in Touch</h4>
        <form id="contact-form" action="#" method="POST" onSubmit={submitQuery}>
          <input
            value={name}
            onChange={(e) => setName(e.target.value)}
            type="text"
            name="name"
            required
            placeholder="Enter Your Name"
          />
          <input
            value={email}
            onChange={(e) => setEmail(e.target.value)}
            type="email"
            name="email"
            required
            placeholder="Enter Your Email"
          />
          <input
            value={subject}
            onChange={(e) => setSubject(e.target.value)}
            type="text"
            name="subject"
            required
            placeholder="Your Subject"
          />
          <textarea
            placeholder="Message"
            required
            name="message"
            value={message}
            onChange={(e) => setMessage(e.target.value)}
          ></textarea>
          <p data-testid='error' className='text-danger font-weight-bold'>{error}</p>
          <p data-testid='msg' className='text-success'>{msg}</p>
          <button type="submit" name="submit">
            Send Message
          </button>
        </form>
      </div>
    </div>
  );
};

export default ContactForm;
