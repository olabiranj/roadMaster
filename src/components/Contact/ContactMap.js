import React from "react";

const ContactMap = () => {
  return (
    <iframe
      title="Andela, Nigeria."
      src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d3963.785387052263!2d3.2345080147708405!3d6.548759595263443!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x103b853fa37c1177%3A0x477d07f40cef86a1!2sOlorunfemi%20Bus%20station!5e0!3m2!1sen!2sng!4v1582307168526!5m2!1sen!2sng"
      width="600"
      height="450"
      frameBorder="0"
      style={{ border: 0, marginBottom: "-10px" }}
      allowFullScreen=""
    ></iframe>
  );
};

export default ContactMap;
