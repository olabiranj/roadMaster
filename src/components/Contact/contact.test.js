import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";
import ContactAddress from "./ContactAddress";
import ContactForm from "./ContactForm";
import ContactMap from "./ContactMap";

describe("contact page", () => {
  afterEach(cleanup);
  it("renders contact address", () => {
    let tree = renderer.create(<ContactAddress />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders contact form with invalid subject error", () => {
    let { getByTestId, getByText, getByPlaceholderText } = render(
      <ContactForm />
    );
    let err = getByTestId("error");
    let errorMsg1 = "*Subject Too Short!";
    expect(err).toHaveTextContent("");
    fireEvent.change(getByPlaceholderText("Enter Your Name"), {
      target: { value: "Joshua" },
    });
    fireEvent.change(getByPlaceholderText("Enter Your Email"), {
      target: { value: "jo@jo" },
    });
    fireEvent.change(getByPlaceholderText("Your Subject"), {
      target: { value: "Hi" },
    });
    fireEvent.change(getByPlaceholderText("Message"), {
      target: { value: "React.js is the most popular frontend framework" },
    });
    fireEvent.click(getByText("Send Message"));
    expect(err).toHaveTextContent(errorMsg1);
  });
  it("renders contact form with invalid message error", () => {
    let { getByTestId, getByText, getByPlaceholderText } = render(
      <ContactForm />
    );
    let err = getByTestId("error");
    let errorMsg2 = "*Message Too Short!";
    expect(err).toHaveTextContent("");
    fireEvent.change(getByPlaceholderText("Enter Your Name"), {
      target: { value: "Joshua" },
    });
    fireEvent.change(getByPlaceholderText("Enter Your Email"), {
      target: { value: "jo@jo" },
    });
    fireEvent.change(getByPlaceholderText("Your Subject"), {
      target: { value: "Hello World" },
    });
    fireEvent.change(getByPlaceholderText("Message"), {
      target: { value: "No message" },
    });
    fireEvent.click(getByText("Send Message"));
    expect(err).toHaveTextContent(errorMsg2);
  });
  it("renders contact form with success message", () => {
    let { getByTestId, getByText, getByPlaceholderText } = render(
      <ContactForm />
    );
    let msg = getByTestId("msg");
    let info = "Message Sent";
    expect(msg).toHaveTextContent("");
    fireEvent.change(getByPlaceholderText("Enter Your Name"), {
      target: { value: "Joshua" },
    });
    fireEvent.change(getByPlaceholderText("Enter Your Email"), {
      target: { value: "jo@jo" },
    });
    fireEvent.change(getByPlaceholderText("Your Subject"), {
      target: { value: "Hello World" },
    });
    fireEvent.change(getByPlaceholderText("Message"), {
      target: {
        value:
          "I have a message and a subject whose length greater than 15 and 5 respectively",
      },
    });
    fireEvent.click(getByText("Send Message"));
    expect(msg).toHaveTextContent(info);
  });
  it("renders contact map", () => {
    let tree = renderer.create(<ContactMap />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
