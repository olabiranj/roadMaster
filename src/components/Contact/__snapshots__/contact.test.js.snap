// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contact page renders contact address 1`] = `
<div
  className="col-lg-6"
>
  <div
    className="contact-address"
  >
    <h4>
      Address
    </h4>
    <p>
      Andela - Lagos Office, EPIC tower, 232 Ikorodu RD, Ilupeju, Lagos.
       
      <br />
      <br />
      <span
        className="text-info"
      >
        You can also reach us on:
      </span>
    </p>
    <ul>
      <li>
        <div
          className="contact-address-icon"
        >
          <i
            className="icofont icofont-headphone-alt"
          />
        </div>
        <div
          className="contact-address-info"
        >
          <a
            href="callto:#"
          >
            +234 8012 435 941
          </a>
          <a
            href="callto:#"
          >
            +234 9041 800 937
          </a>
        </div>
      </li>
      <li>
        <div
          className="contact-address-icon"
        >
          <i
            className="icofont icofont-envelope"
          />
        </div>
        <div
          className="contact-address-info"
        >
          <a
            href="mailto:#"
          >
            team-003gmail.com
          </a>
        </div>
      </li>
      <li>
        <div
          className="contact-address-icon"
        >
          <i
            className="fa fa-github"
          />
        </div>
        <div
          className="contact-address-info"
        >
          <a
            href="#s"
          >
            www.github.com/buildforSDG-003.com
          </a>
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`contact page renders contact map 1`] = `
<iframe
  allowFullScreen=""
  frameBorder="0"
  height="450"
  src="https://www.google.com/maps/embed?pb=!1m16!1m12!1m3!1d30585.49408227924!2d3.366675750650524!3d6.555627266071869!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!2m1!1sandela%20lagos%20map!5e0!3m2!1sen!2sng!4v1588664445446!5m2!1sen!2sng"
  style={
    Object {
      "border": 0,
      "marginBottom": "-10px",
    }
  }
  title="Andela, Nigeria."
  width="600"
/>
`;
